For our Hurd install, assume the GRUB directory is always on the root device.

--- grub-core/osdep/unix/relpath.c
+++ grub-core/osdep/unix/relpath.c
@@ -93,6 +93,7 @@ grub_make_system_path_relative_to_its_ro
       if (stat (buf, &st) < 0)
 	grub_util_error (_("cannot stat `%s': %s"), buf, strerror (errno));
 
+#ifndef __GNU__
       /* buf is another filesystem; we found it.  */
       if (st.st_dev != num)
 	{
@@ -112,6 +113,7 @@ grub_make_system_path_relative_to_its_ro
 	  else
 	    break;
 	}
+#endif
 
       offset = p - buf;
       /* offset == 1 means root directory.  */
